sql: convert LocalOnlySessionData to protobuf

otan opened the pull request:
See individual commits for details.

Refs: #68464

otan committed:
sessiondata: reformat int to int64 for LocalOnlySessionData

This in preparation for the protobuf refactor and making subsequent
reviews easy to read.

Release note: None

otan committed:
sessiondata: convert NoticeDisplaySeverity to uint32

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/68525)
<!-- Reviewable:end -->

otan committed:
sql: convert LocalOnlySessionData to protobuf

This commit converts LocalOnlySessionData to a protobuf. This allows
easy serialization when we implement session migration.

Note there are a few fields remaining that we do not intend to allow
during a session migration -- these have been moved to
`LocalUnmigratableSessionData`.

There are some fields remaining here (enums, sequences) that require
more care and will be moved in a separate commit.

Release note: None

otan committed:
sessiondata: move enums into sessiondatapb

This moves the remaining enums in LocalUnmigratableSessionData to
sessiondatapb.LocalOnlySessionData.

We cannot use native proto enums as the names (OFF, ON, ALWAYS) conflict
with each other in protobufs. As such, resort to casting again.

Release note: None

otan committed:
sessiondata: move LocalOnlySessionData enums to sessiondatapb

This is in preparation for enum-ifying these variables.
All changes applied here are through Goland refactor.

Release note: None

rafiss reviewed (approved):
this lgtm! i actually think the SequenceCache stuff might never need to be done, since CACHED sequences already do not guarantee zero gaps in the sequence increment

otan commented:
fixed up, tftr!

bors r=rafiss
